validator/parse-css.js

```diff
@@ -521,37 +153,21 @@ function consumeAFunction(tokenStream) {
  */
 parse_css.parseAStylesheet = function(
     tokenList, atRuleSpec, defaultSpec, errors) {
-  /** @type {!parse_css.TokenStream} */
-  const tokenStream = new parse_css.TokenStream(tokenList);
-  /** @type {!parse_css.Stylesheet} */
-  const sheet = new parse_css.Stylesheet();
-  sheet.value = consumeAListOfRules(tokenStream, /*topLevel=*/true, errors);
-  sheet.line = sheet.value[0].line;
-  sheet.col = sheet.value[0].col;
+  /** @type {!Canonicalizer} */
+  const canonicalizer = new Canonicalizer(atRuleSpec, defaultSpec);
 
-  return parse_css.canonicalize(sheet, atRuleSpec, defaultSpec, errors);
-};
+  /** @type {!parse_css.Stylesheet} */
+  const stylesheet = new parse_css.Stylesheet();
 
-/**
- * @param {!Array<!parse_css.CSSParserToken>} tokenList
- * @param {!Array<!parse_css.ErrorToken>} errors output array for the errors.
- * @return {!Array<!parse_css.CSSParserRule>}
- */
-parse_css.parseAListOfRules = function(tokenList, errors) {
-  /** @type {!parse_css.TokenStream} */
-  const tokenStream = new parse_css.TokenStream(tokenList);
-  return consumeAListOfRules(tokenStream, /*topLevel=*/false, errors);
-};
+  stylesheet.rules = canonicalizer.parseAListOfRules(
+      tokenList, /*topLevel=*/true, errors);
+  stylesheet.line = tokenList[0].line;
+  stylesheet.col = tokenList[0].col;
+  const eof = /** @type {parse_css.EOFToken} */
```

Comment:
It's guaranteed that the lat token is EOFToken? If so, I assume it's probably non-nullable? If so, please add `!`